From f9bbd8636a7a77c94987a9b89da85933e1ec8b09 Mon Sep 17 00:00:00 2001
From: Stefan Tatschner <stefan@sevenbyte.org>
Date: Sat, 14 Mar 2015 19:02:38 +0100
Subject: [PATCH 1/2] systemd: ProtectHome was too strict

"ProtectHome=true" was too strict and caused problems when running under
the system instance. The problem does not appear when syncthing-inotify
is running under the user instance, thus I missed it in the first place.

"ProtectHome=read-only" fixes the problem.

Reported by: madalu <mdl@imapmail.org>
Reported on: https://aur.archlinux.org/packages/syncthing-inotify/
---
 etc/linux-systemd/system/syncthing-inotify@.service | 2 +-
 etc/linux-systemd/user/syncthing-inotify.service    | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/etc/linux-systemd/system/syncthing-inotify@.service b/etc/linux-systemd/system/syncthing-inotify@.service
index c198a5b..3471ad0 100644
--- a/etc/linux-systemd/system/syncthing-inotify@.service
+++ b/etc/linux-systemd/system/syncthing-inotify@.service
@@ -10,7 +10,7 @@ ExecStart=/usr/bin/syncthing-inotify
 Restart=on-failure
 PrivateTmp=true
 ProtectSystem=full
-ProtectHome=true
+ProtectHome=read-only
 
 [Install]
 WantedBy=multi-user.target
diff --git a/etc/linux-systemd/user/syncthing-inotify.service b/etc/linux-systemd/user/syncthing-inotify.service
index 0afcd61..42df174 100644
--- a/etc/linux-systemd/user/syncthing-inotify.service
+++ b/etc/linux-systemd/user/syncthing-inotify.service
@@ -9,7 +9,7 @@ ExecStart=/usr/bin/syncthing-inotify
 Restart=on-failure
 PrivateTmp=true
 ProtectSystem=full
-ProtectHome=true
+ProtectHome=read-only
 
 [Install]
 WantedBy=default.target

From 00ef95b2947af54a2409d5f7dbb1219a9aca94c9 Mon Sep 17 00:00:00 2001
From: Stefan Tatschner <stefan@sevenbyte.org>
Date: Sun, 15 Mar 2015 15:24:54 +0100
Subject: [PATCH 2/2] systemd: "PrivateTmp" is too strict

We decided to drop "PrivateTmp" because syncthing-inotify does not write
to "/tmp". Furthermore if somebody decides to watch "/tmp" odd things
will happen as "PrivateTmp" mounts an isolated "/tmp" instance for
the service.
---
 etc/linux-systemd/system/syncthing-inotify@.service | 1 -
 etc/linux-systemd/user/syncthing-inotify.service    | 1 -
 2 files changed, 2 deletions(-)

diff --git a/etc/linux-systemd/system/syncthing-inotify@.service b/etc/linux-systemd/system/syncthing-inotify@.service
index 3471ad0..dab2e2f 100644
--- a/etc/linux-systemd/system/syncthing-inotify@.service
+++ b/etc/linux-systemd/system/syncthing-inotify@.service
@@ -8,7 +8,6 @@ Requires=syncthing@.service
 User=%i
 ExecStart=/usr/bin/syncthing-inotify
 Restart=on-failure
-PrivateTmp=true
 ProtectSystem=full
 ProtectHome=read-only
 
diff --git a/etc/linux-systemd/user/syncthing-inotify.service b/etc/linux-systemd/user/syncthing-inotify.service
index 42df174..780f0cb 100644
--- a/etc/linux-systemd/user/syncthing-inotify.service
+++ b/etc/linux-systemd/user/syncthing-inotify.service
@@ -7,7 +7,6 @@ Requires=syncthing.service
 [Service]
 ExecStart=/usr/bin/syncthing-inotify
 Restart=on-failure
-PrivateTmp=true
 ProtectSystem=full
 ProtectHome=read-only
 
